libs/product/state/src/selectors/composite-product-entities/composite-product-entities.selectors.ts

```diff
@@ -22,12 +22,38 @@ import { DaffProductReducersState } from '../../reducers/product-reducers-state.
 import { getDaffProductEntitiesSelectors } from '../product-entities/product-entities.selectors';
 import { getDaffProductFeatureSelector } from '../product-feature.selector';
 
+/**
+ * An interface for selectors related to the composite product applied options.
+ */
 export interface DaffCompositeProductEntitiesMemoizedSelectors {
+	/**
+	 * The rxjs entities state for composite product applied options.
+	 */
 	selectCompositeProductAppliedOptionsEntitiesState: MemoizedSelector<Record<string, any>, EntityState<DaffCompositeProductEntity>>;
+	/**
+	 * A selector for all composite product ids in state.
+	 */
 	selectCompositeProductIds: MemoizedSelector<Record<string, any>, EntityState<DaffCompositeProductEntity>['ids']>;
+	/**
+	 * The rxjs entities for the composite product appllied options.
+	 */
 	selectCompositeProductAppliedOptionsEntities: MemoizedSelector<Record<string, any>, EntityState<DaffCompositeProductEntity>['entities']>;
+	/**
+	 * The total number of composite products in state.
+	 */
 	selectCompositeProductTotal: MemoizedSelector<Record<string, any>, number>;
+	/**
+	 * Selects the applied options for a composite product.
+	 *
+	 * @param id the id of the composite product.
+	 */
 	selectCompositeProductAppliedOptions: (id: DaffCompositeProduct['id']) => MemoizedSelector<Record<string, any>, Dictionary<DaffCompositeProductItemOption>>;
+	/**
+	 * Selects whether the particular item of a composite product is required in order for the product to be valid, i.e. addable to the cart.
+	 *
+	 * @param id the id of the composite product.
```

Comment:
duplicate @param docs?

Reply:
No, the first is the product id. The second is the product item id